gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/ReevaluateDagProc.java

```diff
@@ -90,9 +70,28 @@ protected void act(DagManagementStateStore dagManagementStateStore, Pair<Optiona
     }
 
     Dag.DagNode<JobExecutionPlan> dagNode = dagNodeWithJobStatus.getLeft().get();
-    JobStatus jobStatus = dagNodeWithJobStatus.getRight().get();
-    ExecutionStatus executionStatus = dagNode.getValue().getExecutionStatus();
+
+    if (!dagNodeWithJobStatus.getRight().isPresent()) {
+      // Usually reevaluate dag action is created by JobStatusMonitor when a finished job status is available,
+      // but when reevaluate/resume/launch dag proc found multiple parallel jobs to run next, it creates reevaluate
+      // dag actions for each of those parallel job and in this scenario there is no job status available.
+      // If the job status is not present, this job was never launched, submit it now.
+      DagProcUtils.submitJobToExecutor(dagManagementStateStore, dagNode, getDagId());
+      return;
+    }
+
     Dag<JobExecutionPlan> dag = dagManagementStateStore.getDag(getDagId()).get();
+    JobStatus jobStatus = dagNodeWithJobStatus.getRight().get();
+    ExecutionStatus executionStatus = ExecutionStatus.valueOf(jobStatus.getEventName());
+    setStatus(dagManagementStateStore, dagNodeWithJobStatus.getLeft().get(), executionStatus);
+
+    if (!FlowStatusGenerator.FINISHED_STATUSES.contains(executionStatus.name())) {
+      // this may happen if adding job status in the store failed after adding a ReevaluateDagAction in KafkaJobStatusMonitor
+      throw new RuntimeException(String.format("Job status for dagNode %s is %s. Re-evaluate dag action are created for"
```

Comment:
there used to be a `log.warn` in addition to the `RuntimeException`.  seems like worth continuing unless you have a specific objection

Reply:
it was the exact message right? this exception will also be logged right?